pensieve/tests/integration/test_analysis_integration.py

```diff
@@ -0,0 +1,142 @@
+import datetime as dt
+from google.cloud import bigquery
+import mock
+import os
+import pytz
+import pytest
+
+from mozanalysis.metrics import Metric, DataSource, agg_sum
+from google.api_core.exceptions import NotFound
+from pensieve.analysis import Analysis, BigQueryClient
+from pensieve.experimenter import Experiment, Variant
+
+
+@pytest.mark.integration
+class TestAnalysisIntegration:
+    project_id = os.getenv("GOOGLE_PROJECT_ID")
+
+    if project_id is None:
+        print("GOOGLE_PROJECT_ID is not set.")
+        assert False
+
+    test_project = "pensieve-integration-test"
+    test_dataset = "test"
+    # contains the tables filled with test data required to run metrics analysis
+    static_dataset = "test_data"
+
+    client = bigquery.client.Client(project_id)
+
+    clients_daily_source = "pensieve/tests/data/test_clients_daily.ndjson"
+    events_source = "pensieve/tests/data/test_events.ndjson"
+    get_key_udf = "pensieve/tests/data/get_key.sql"
+
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        # remove all tables previously created
+        try:
+            self.client.delete_dataset(self.test_dataset, delete_contents=True, not_found_ok=True)
+            self.client.create_dataset(self.test_dataset)
+        except NotFound:
+            pass
+
+        try:
+            self.client.get_table(f"{self.static_dataset}.clients_daily")
+        except NotFound:
+            table_ref = self.client.create_table(f"{self.static_dataset}.clients_daily")
+            job_config = bigquery.LoadJobConfig()
+            job_config.source_format = bigquery.SourceFormat.NDJSON
+            job_config.autodetect = True
+
+            with open(self.clients_daily_source, "rb") as source_file:
+                job = self.client.load_table_from_file(
+                    source_file, table_ref, job_config=job_config
+                )
+
+            job.result()  # Waits for table load to complete.
+
+        try:
+            self.client.get_table(f"{self.static_dataset}.events")
+        except NotFound:
+            table_ref = self.client.create_table(f"{self.static_dataset}.events")
+            job_config = bigquery.LoadJobConfig()
+            job_config.source_format = bigquery.SourceFormat.NDJSON
+            job_config.autodetect = True
+
+            with open(self.events_source, "rb") as source_file:
+                job = self.client.load_table_from_file(
+                    source_file, table_ref, job_config=job_config
+                )
+
+            job.result()  # Waits for table load to complete.
+
+        # create required UDF
+        with open(self.get_key_udf, "r") as source_file:
+            self.client.query(source_file.read()).result()
+
+    def bq_execute(cls, *args, **kwargs):
+        # to use the test project and dataset, we need to change the SQL query
+        # generated by mozanalysis
```

Comment:
What are the references to prod data? Is it just the UDF? Otherwise this seems like a bug that we shouldn't try to mock around -- it seems a little sad to have to rewrite the client code that I think we'd like to be able to test here.

Reply:
The main issue here is `moz-fx-data-shared-prod.telemetry.events` which is used in the mozanalysis query with the project specified. But yeah, also the UDF since the test environment doesn't have access to `moz-fx-data-shared-prod`

Reply:
I think if we wanted to fully test the client code then maybe `client.query()` could be mocked out instead but this wouldn't be much different as the values for the sql `query` and `config` would need to be modified anyways. 

Perhaps mozanalysis can make the project and dataset configurable?